pages/todosView.vue

```diff
@@ -1,18 +1,22 @@
 <script setup>
 import { ref } from "vue";
-import { storeToRefs } from "pinia";
-import { useUserStore } from "../stores/userStore";
-import TodoCard from "../components/todos/TodoCard.vue";
-
-const userStore = useUserStore();
-const { todos } = storeToRefs(userStore);
+import { useQuery } from "@tanstack/vue-query";
+import TodoCard from "@/components/todos/TodoCard.vue";
+import { api } from "@/utils/api";
 
 const newTodo = ref("");
 
 const submitHandler = () => {
     console.log("todo submitted");
     newTodo.value = "";
 };
+
+const { data } = useQuery({
+    queryKey: ["todos"],
+    queryFn: async () => {
+        return api.get("todos");
+    },
+});
```

Comment:
Préfère la syntaxe "simple" :
```suggestion
const { data } = useQuery(["todos"], () => {
    return api.get("todos");
});
```

Tu n'as également pas besoin de déclarer ta fonction en `async` car tu n'utilises pas d'`await`. C'est une simple fonction qui retourne une Promise.

Reply:
J'ai voulu utiliser cette syntaxe mais elle me renvoie une erreur.